src/MongoDB.Driver.Core/Core/Operations/MapReduceOperationBase.cs

```diff
@@ -120,6 +117,7 @@ public BsonJavaScript FinalizeFunction
         /// </remarks>
         ///   <c>true</c> if objects emitted by the map function remain as JavaScript objects; otherwise, <c>false</c>.
         /// </value>
+        [Obsolete("JavaScriptMode is ignored by server versions 4.3.1 and newer.")]
```

Comment:
According to driver ticket, this option will be ignored:
>The following will be ignored in 4.4

>    jsMode: Specifies whether to convert intermediate data into BSON format between the execution of the map and reduce functions. There will be no option to maintain intermediate objects as JavaScript objects.

Reply:
Do we want to use `4.4.0` in these user facing messages?

I realize 4.3.1 may be technically correct, but very few users use pre-release versions of the server and may find this version number confusing.

Reply:
I copied this message from your PR with deprecating fields :) And I had the same suggestion for your PR, but then I decided that the current way is also good enough. So, I will be ok with any message, but if we decide to change it here, I will suggest the same change in your PR.

Reply:
@jyemin , what do you think about it? Is it a good enough if we add the pre-release server version to the obsolete message? Should we worry about the case when not all users understand that 4.3.1 is a pre-release version of 4.4.0.?

Reply:
Definitely use 4.4.0.  I typically just say 4.4 and don't mention the patch level.

Reply:
thanks

Reply:
done